fdbserver/ConfigBroadcaster.actor.cpp

```diff
@@ -0,0 +1,476 @@
+/*
+ * ConfigBroadcaster.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <algorithm>
+
+#include "fdbserver/ConfigBroadcaster.h"
+#include "fdbserver/Knobs.h"
+#include "fdbserver/IConfigConsumer.h"
+#include "flow/UnitTest.h"
+#include "flow/actorcompiler.h" // must be last include
+
+namespace {
+
+bool matchesConfigClass(ConfigClassSet const& configClassSet, Optional<KeyRef> configClass) {
+	return !configClass.present() || configClassSet.contains(configClass.get());
+}
+
+// Helper functions for STL containers, with flow-friendly error handling
+template <class MapContainer, class K>
+auto get(MapContainer&& m, K const& k) -> decltype(m.at(k)) {
+	auto&& mapContainer = std::forward<MapContainer>(m);
```

Comment:
Is this line necessary? My knowledge here is not great, but why can't you just use `m` in place of `mapContainer` below? To me, it looks like both are of type `MapContainer&&`.

Reply:
You're right, I'll simplify these helper functions.